src/platform-aws.c

```diff
@@ -519,6 +539,12 @@ int platform_init(const char **provider_filter)
 		nccl_ofi_selected_protocol = platform_data->default_protocol;
 	}
 
+	domain_per_endpoint = ofi_nccl_domain_per_endpoint();
+	if (domain_per_endpoint == -1) {
+		domain_per_endpoint = platform_data->domain_per_endpoint;
+	}
+	NCCL_OFI_INFO(NCCL_INIT | NCCL_NET, "Creating one domain per %s", domain_per_endpoint ? "endpoint" : "device");
```

Comment:
Nitpick: Just this commit by itself does not change how the domain is created, it just introduces the config. The print should come in the second commit.